test/utils/searcher/search.go

```diff
@@ -0,0 +1,98 @@
+package searcher
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/amp-labs/connectors/test/utils"
+	"github.com/spyzhov/ajson"
+)
+
+type Type string
+
+const (
+	Object  Type = "object"
+	Array   Type = "array"
+	String  Type = "string"
+	Integer Type = "integer"
+)
+
+type Key struct {
+	Type  Type
+	At    string
+	Index int // relevant if type is array
+}
+
+func Find[T any](res *common.ReadResult, keys []Key, value T) map[string]any {
```

Comment:
It is a utility function for `test` folder.
It helps locating the `Read Row` that matches the `value` given compound list of keys.